src/domain/끄적이는/components/EditInput.tsx

```diff
@@ -0,0 +1,78 @@
+import { type ChangeEvent, type KeyboardEvent, useRef, useState } from 'react';
+
+import { type UseInputReturn } from '@hooks/useInput';
+
+import * as styles from './editInput.css';
+
+interface EditInputProps {
+  inputProps: UseInputReturn;
+}
+
+const EditInput = ({ inputProps }: EditInputProps) => {
+  const inputRef = useRef<HTMLTextAreaElement | null>(null);
+  const [textareaHeight, setTextareaHeight] = useState<{
+    row: number;
+    lineBreak: Record<number, boolean>;
+  }>({
+    row: 1,
+    lineBreak: {},
+  });
+
+  const handleResize = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    const { scrollHeight, clientHeight, value } = e.target;
+
+    if (value.length === 0) {
+      setTextareaHeight((prev) => ({
+        row: 1,
+        lineBreak: { ...prev.lineBreak, [e.target.value.length]: false },
+      }));
+    }
+
+    if (scrollHeight > clientHeight) {
+      setTextareaHeight((prev) => ({
+        row: prev.row + 1,
+        lineBreak: { ...prev.lineBreak, [value.length - 1]: true },
+      }));
+    }
+
+    if (textareaHeight.lineBreak[value.length]) {
+      setTextareaHeight((prev) => ({
+        row: prev.row - 1,
+        lineBreak: { ...prev.lineBreak, [value.length]: false },
+      }));
+    }
+  };
+
+  const handleKeydownEnter = (e: KeyboardEvent<HTMLTextAreaElement>) => {
+    if (e.code === 'Enter') {
+      setTextareaHeight((prev) => ({
+        row: prev.row + 1,
+        lineBreak: { ...prev.lineBreak, [inputProps.value.length]: true },
+      }));
+      return;
+    }
+  };
```

Comment:
```suggestion
  const handleKeydown = (e: KeyboardEvent<HTMLTextAreaElement>) => {
    switch (e.code) {
      case 'Enter':
       setTextareaHeight((prev) => ({
        row: prev.row + 1,
        lineBreak: { ...prev.lineBreak, [inputProps.value.length]: true },
      }));
      return;
    }
  };
```
추후에 여러 키에 대해서 리팩토링해봐도 좋을 듯 하네요